bot.ml

```diff
@@ -786,15 +786,158 @@ let callback _conn req body =
+                      ~id:pr.id
+                | None -> (
+                    if
+                      not
+                        (List.exists comment_info.issue.labels ~f:(fun label ->
+                             string_match ~regexp:"kind:.*" label))
+                    then
+                      GitHub_mutations.post_comment ~bot_info
+                        ~message:
+                          (f "@%s: No kind label found, merge aborted."
+                             comment_info.author)
+                        ~id:pr.id
+                    else if not comment_info.issue.milestoned then
+                      GitHub_mutations.post_comment ~bot_info
+                        ~message:
+                          (f "@%s: No milestone found, merge aborted."
+                             comment_info.author)
+                        ~id:pr.id
+                    else
+                      match reviews_info.review_decision with
+                      | REVIEW_REQUIRED | CHANGES_REQUESTED ->
+                          GitHub_mutations.post_comment ~bot_info
+                            ~message:
+                              (f
+                                 "@%s: PR cannot be merged as it isn't \
+                                  approved yet."
+                                 comment_info.author)
+                            ~id:pr.id
+                      | APPROVED -> (
+                          if
+                            not
+                              (List.exists comment_info.issue.assignees
+                                 ~f:(fun login ->
+                                   String.equal login comment_info.author))
+                          then
+                            GitHub_mutations.post_comment ~bot_info
+                              ~message:
+                                (f
+                                   "@%s: You can't merge the PR as you're not \
+                                    among the assignees."
+                                   comment_info.author)
+                              ~id:pr.id
+                          else if String.equal comment_info.author pr.user then
+                            GitHub_mutations.post_comment ~bot_info
+                              ~message:
+                                (f
+                                   "@%s: You can't merge the PR since you \
+                                    authored it."
+                                   comment_info.author)
+                              ~id:pr.id
+                          else if
+                            not (String.equal reviews_info.baseRef "master")
+                          then
+                            GitHub_mutations.post_comment ~bot_info
+                              ~message:
+                                (f
+                                   "@%s: PR targets branch %s instead of \
+                                    master, abort merge."
```

Comment:
```suggestion
                                   "@%s: This PR targets branch `%s` instead of \
                                    `master`. Only release managers can merge in \
                                    release branches. Merging with the bot is not \
                                    supported."
```